frontend/src/store/transaction.js

```diff
@@ -49,30 +50,84 @@ export default ({
       context.commit('setUser', response.data)
     },
     getRegisteredNodes (context, params) {
-      let page = params.page || context.state.nodePage
+      let page = context.state.nodePage
       if (!page) return
 
-      tfService.getNodes(undefined, params.size, page).then(response => {
+      const getNodes = (page) => {
+        return tfService.getNodes(undefined, params.size, page)
+      }
+
+      getNodes(page).then(response => {
         context.commit('setRegisteredNodes', response)
-        context.commit('setTotalSpecs', response.data)
+        const pages = parseInt(response.headers.pages, 10)
+        const promises = []
+        for (var index = context.state.nodePage; index <= pages; index++) {
+          promises.push(getNodes(index))
+        }
+        Promise.all(promises)
+          .then(res => {
+            res.map(response => {
+              context.commit('setRegisteredNodes', response)
+            })
+          })
+          .finally(() => {
+            context.commit('setTotalSpecs', context.state.registeredNodes)
+            context.commit('setNodesLoading', false)
+          })
       })
     },
     getRegisteredFarms (context, params) {
-      let page = params.page || context.state.farmPage
+      let page = context.state.nodePage
       if (!page) return
 
-      tfService.registeredfarms(params.size, page).then(response => {
-        context.commit('setAmountOfFarms', response.data)
+      const getFarms = (page) => {
+        return tfService.registeredfarms(params.size, page)
+      }
+
+      getFarms(page).then(response => {
         context.commit('setRegisteredFarms', response)
+        const pages = parseInt(response.headers.pages, 10)
+        const promises = []
+        for (var index = context.state.farmPage; index <= pages; index++) {
+          promises.push(getFarms(index))
+        }
+        Promise.all(promises)
+          .then(res => {
+            res.map(response => {
+              context.commit('setRegisteredFarms', response)
+            })
+          })
+          .finally(() => {
+            context.commit('setAmountOfFarms', context.state.registeredFarms)
+            context.commit('setFarmsLoading', false)
+          })
       })
     },
     getRegisteredGateways (context, params) {
       let page = params.page || context.state.gatewayPage
```

Comment:
this will always be 1 here no ? since we always get all the gateway ?

Reply:
No, it will increment if there are more pages https://github.com/threefoldtech/tfexplorer/blob/b2b43649918dd387125ee886485db85bbf08f05e/frontend/src/store/transaction.js#L173